palo_alto_pan_os/komand_palo_alto_pan_os/util/pan_os_requests.py

```diff
@@ -18,17 +19,20 @@ def new_session(cls, connection: Connection, username: str, password: str, hostn
         url = hostname + "/api/"
         querystring = {"type": "keygen", "user": username, "password": password}
 
+        token_response = {"text": ""}
         try:
             session = requests.session()
             token_response = session.get(url, params=querystring, verify=verify_cert)
             token_dict = xmltodict.parse(token_response.text)
-            try:
-                pan_os_key = token_dict["response"]["result"]["key"]
-                connection.logger.info("Key obtained")
-            except KeyError:
-                raise ConnectionTestException(
-                    preset=ConnectionTestException.Preset.USERNAME_PASSWORD)
-
+            pan_os_key = token_dict["response"]["result"]["key"]
+            connection.logger.info("Key obtained")
+        except KeyError:
+            raise ConnectionTestException(
+                preset=ConnectionTestException.Preset.USERNAME_PASSWORD
+            )
+        except requests.exceptions.HTTPError as e:
+            connection.logger.info(f"Call to Palo Alto Firewall API failed: {e}")
+            raise ConnectionTestException(preset=PluginException.Preset.UNKNOWN, data=token_response.text)
```

Comment:
Can we be more specific here? 

for instance, can we do anything with the status code? (500 - server error, 404 - unauthorized, etc...)